src/IO/GEHDF5Wrapper.cxx

```diff
@@ -283,18 +284,29 @@ Succeeded GEHDF5Wrapper::initialise_singles_data(const std::string &path)
 {
     if(path.size() == 0)
     {
-        if(is_signa)
+        if(is_signa) //AB todo "is_RDF9"
         {
-            m_address = "/Singles/CrystalSingles/sample";
-            //! \todo Get these numbers from the HDF5 file
-            {
-                m_NX_SUB = 45;    // hyperslab dimensions
-                m_NY_SUB = 448;
-                m_NZ_SUB = 1;
-                m_NX = 45;        // output buffer dimensions
-                m_NY = 448;
-                m_NZ = 1;
-            }
+            m_address = "/Singles/CrystalSingles/sample1"; //AB: todo. '/slms/inm/research/moco/PhantomData_Orsay/R00037/raw/rdf.0' had ./sample1 but ./sample was written here. ?? typo?
+            // Get the DataSpace (metadata) corresponding to the DataSet that we want to read
+            m_dataset_sptr.reset(new H5::DataSet(file.openDataSet(m_address)));
+            m_dataspace = m_dataset_sptr->getSpace();
+            // Create an array to host the size of the dimensions
+            const int rank = m_dataspace.getSimpleExtentNdims();
+            hsize_t dims[rank];
+            // hsize_t max_dims[dataspace_Ndims]; // AB: Do we want the max_dims?
+            // Read size of dimensions
+            m_dataspace.getSimpleExtentDims( dims, NULL); // AB: Do we want the max_dims?
+
+            // AB todo: I think it returs dataspace_Ndims==2, test.
+            // AB todo: MATLAB reader returns [448 45] yet here it was written NX=45 NY=448. row/colun major issue, or actual difference? test
+
```

Comment:
MATLAB may have transposed the row and column so I would suggest to make sure if everything is read in correct order.

Reply:
@pw351 indeed MATLAB transposes so it can read in column major order. I am changing all these after testing.